MCMC/simulation.py

```diff
@@ -53,18 +59,46 @@ def trial_move(self):
                 generate a random number p'
                 if p' < p : Accept else reject
         """
-        return NotImplementedError
+        _system = np.copy(self.system)
+        # Select random occupied site
+        non_zero_indices = list(np.nonzero(self.system))
+        random.shuffle(non_zero_indices)
+        site1 = non_zero_indices[0]
+        site1_val = _system[site1]
+        # Select random site from all possible sites
+        all_indices = list(self.grid_indices) 
+        random.shuffle(all_indices)
+        site2 = all_indices[0]
+        site2_val = _system[site2]
+        # Swap the values of each site and find change in energy
+        _system[site1] = site2_val
+        _system[site2] = site1_val
+        delta_U = self.calculate_energy(_system) - self.energy
+
+        if delta_U <= 0:
+            self.system = _system
+            self.accepted_moves += 1
+        else:
+            rand_num = random.uniform(0, 1)
+            if np.exp(-delta_U/self.kT) <= rand_num:
+                self.system = _system
+                self.accepted_moves += 1
+            else:
+                pass
```

Comment:
I'd suggest moving accept/reject section to a new function that takes delta_U as an argument. We might develop multiple versions of trial moves but the acceptance criteria will remain the same for all.